### ISSUE TITLE
Option to "Disable code optimization"

### ISSUE THREAD

**Issue by pedroflor** [Col-E/Recaf#246]
## Option to "Disable code optimization"

Maybe something like **Config -> Assembler => Disable code optimization**

The main idea is to have an option to "**disable**" code optimization when modifying a class.

Please refer to the following images for a better explanation.

**Original code with the instruction  `IFNONNULL J` replaced with `GOTO T`**
![image](https://user-images.githubusercontent.com/3332674/91092987-c961d880-e626-11ea-8390-d742e573b9ea.png)

**Resulting method after the optimization**
![image](https://user-images.githubusercontent.com/3332674/91093027-d4b50400-e626-11ea-9284-1621cd9a5dfb.png)

Recaf-2.4.1

**Comment by xxDark** [comment]
That can't be turned off.
ASM replaces dead code with nops plus athrow in order to generate valid stack map.

**Comment by pedroflor** [comment]
Thanks for the info.

**Comment by Col-E** [comment]
Yeah, it would be nice, but this is an ASM _feature_: [MethodWriter.java](https://gitlab.ow2.org/asm/asm/-/blob/master/asm/src/main/java/org/objectweb/asm/MethodWriter.java#L1624)

Like most things in ASM, its code inside a private method and is tightly coupled with other private logic, so it can't be easily removed by extension _(This is a common head-ache when attempting to mess with adding behavior to ASM)_. Great library, but the developers don't intend on making large portions of their code extensible.

I'd be happy to accept a PR if somebody finds a magical _(and clean)_ way to resolve this. I would prefer to not ship our own fork of ASM.

**Comment by Dusto-beep** [comment]
I'm lost here

**Comment by Dusto-beep** [comment]
So the verdict bitches where's the laws can't be broken but where there is no law at all to be broken bugs we fixed stomps whatever it is behind and we or us or just me right back ahead into the very first place I've mined you now its all in data backed also from day time ones very first start after mine when you was ahead at one time but fallen back down way behind the true block my friends thanks I've can retry to my own apps and creations buddy!!
